script.js

```diff
@@ -1,12 +1,440 @@
+    episodeImageHtml.classList.add("medium-img");
+    episodeImageContainerHtml.classList.add("episode-image-container");
+    episodeImageHtml.src = episodeImageSrc;
+
+    // Get the episode summary
+    const episodeSummaryTextHtml = document.createElement("p");
+    const episodeSummaryContainerHtml = document.createElement("div");
+    episodeSummaryTextHtml.classList.add("episode-summary-text");
+    episodeSummaryContainerHtml.classList.add("episode-summary-container");
+    const episodeSummary = episode.summary;
+    episodeSummaryTextHtml.innerHTML = episodeSummary;
+
+    // Update info about displayed episodes
+    episodeSearchResultDisplayHtml.textContent = `Displaying ${index + 1}/${
+      episodeList.length
+    } episodes`;
+
+    // Add the episode card elements to the container
+    episodeNameNumSeasonComboHolderHtml.appendChild(
+      episodeNameNumSeasonComboTextHtml
+    );
+    episodeCardHtml.appendChild(episodeNameNumSeasonComboHolderHtml);
+    episodeCardHtml.appendChild(episodeImageContainerHtml);
+    episodeImageContainerHtml.appendChild(episodeImageHtml);
+    episodeSummaryContainerHtml.appendChild(episodeSummaryTextHtml);
+    episodeCardHtml.appendChild(episodeSummaryContainerHtml);
+    cardsContainerHtml.appendChild(episodeCardHtml);
+    rootHtml.appendChild(cardsContainerHtml);
+  }); // end of for.each
 }
 
-window.onload = setup;
+// level 200
+
+// function to search episodes that matches with inputValue
+
+function displaySearchedEpisodes(inputValue) {
+  // creating a fresh array which only holds the episodes that match our search criteria
+  const filteredEpisodes = allEpisodes.filter((episode) => {
+    if (episode.summary !== null && episode.name !== null) {
+      return (
+        episode.name.toLowerCase().includes(inputValue) ||
+        episode.summary.toLowerCase().includes(inputValue)
+      );
+    }
+  });
+
+  // Display searchedEpisodes
+  rootHtml.innerHTML = "";
+  makePageForEpisodes(filteredEpisodes);
+  // makePageForEpisodes(episodeList);
+  buildEpisodeDropdownList(filteredEpisodes);
+
+  episodeSearchResultDisplayHtml.textContent = `Displaying ${filteredEpisodes.length}/${allEpisodes.length} episodes`;
+}
+
+// moved the eventListener to the bottom to level 500
+
+// level 300
+
+// jump to episode function
+
+function jumpToEpisode(event) {
+  console.log("JumpToEpisode event.target.value", event.target.value);
+  const episodeSelectHtml = document.querySelector("#episode-select-html"); // this is the episode dropdown element
+
+  const selectedEpisode = allEpisodes.filter((episode) => {
+    // console.log(episode.name);
+    // console.log(episodeSelectHtml.value);
+    return episode.name === episodeSelectHtml.value;
+  });
+  console.log(selectedEpisode, "<---selectedEpisode");
+  // Display selectedEpisode
+  rootHtml.innerHTML = "";
+  makePageForEpisodes(selectedEpisode);
+  // buildEpisodeDropdownList(selectedShows);
```

Comment:
I'd recommend deleting the commented  code, it makes it easier to follow what's actually going on (and easier to review!)

Reply:
Apologies for that. I have removed those comments and the unnecessary console logs.